### DISCUSSION TITLE
Example usage?

### DISCUSSION THREAD

**Discussion by kaminskyalexander** [trentschnee/gatsby-theme-joy-ui#11] [Q&A] [answered]
Hi! This is my first time using Gatsby and I'm wondering if you could provide an example of how to use this plugin? Looking in the source code, it looks as if there is a component which is supposed to wrap the page with `CssVarsProvider` and `CssBaseline` however I am unsure of how to get this to work. If I don't wrap the page manually myself with these components I get an error.

**Comment by trentschnee** [comment, marked as answer]
@alexkmnsky Hey alexkmnsky, thanks for using my plugin. I think that is a very great suggestion and I have [created an issue](https://github.com/trentschnee/gatsby-theme-joy-ui/issues/12) to come up with some examples. Quick question, do you have any type of specific example you are wanting?

**Reply by kaminskyalexander** [reply]
Thank you for making this plugin! A good one could be from the getting started tutorial: https://mui.com/joy-ui/getting-started/tutorial/

My specific case is that I would like to use the `Stack` component, however it has issues with `style` tags being injected inside of the body next to each element on the static pages generated by Gatsby, which causes issues with spacing. I have found online suggestions to wrap the page with `StyledEngineProvider`, passing in the `injectFirst` prop [like this](https://mui.com/material-ui/guides/interoperability/#css-injection-order) in order to place the styles in the head, which did not work for me. I am unsure if this is an issue with the plugin though.

Here's an example which reproduces this:
```jsx
<Stack spacing={1}>
  <Typography level="h2" fontSize="xl2">
    Example
  </Typography>
  <Typography level="body3" sx={{ fontWeight: "md", color: "text.secondary" }}>
    Spacing issue
  </Typography>
</Stack>
```

Javascript disabled:
![image](https://user-images.githubusercontent.com/29873742/210031449-0c1680cb-9c49-4ff9-b6fe-8cbfe4a43e0e.png)

Javascript enabled:
![image](https://user-images.githubusercontent.com/29873742/210031513-e9c66e2f-051a-4066-9589-6b77be9369b8.png)

(the spacing is different in the two screenshots and the cause is that the placement of the style element in the static page)

**Reply by trentschnee** [reply]
Just to update you: I've been working on refactoring this into a monorepo with examples but I am taking the time to do it the right way. Thank you for your patience and shedding light on this issue!

**Reply by trentschnee** [reply]
Hey @alexkmnsky . Just letting you know that I finished my refactoring with [MUI Joy's Introductory tutorial](https://mui.com/joy-ui/getting-started/tutorial/) example.

See Completed in release [1.2.0](https://github.com/trentschnee/gatsby-theme-joy-ui/releases/tag/v1.2.0)

**Reply by kaminskyalexander** [reply]
Awesome, just checked it out! 

Thank you so much!

**Reply by kaminskyalexander** [reply]
Hey @trentschnee

I cloned the latest release and copied the introductory tutorial example into a separate directory. After running `npm install` and `npm run develop` I am getting this error:
```
Error in function useColorScheme in ./node_modules/@mui/system/esm/cssVars/createCssVarsProvider.js:37

MUI: `useColorScheme` must be called under <CssVarsProvider />
./node_modules/@mui/system/esm/cssVars/createCssVarsProvider.js:37

  35 |     const value = React.useContext(ColorSchemeContext);
  36 |     if (!value) {
> 37 |       throw new Error(process.env.NODE_ENV !== "production" ? `MUI: \`useColorScheme\` must be called under <CssVarsProvider />` : _formatMuiErrorMessage(19));
  38 |     }
  39 |     return value;
  40 |   };
```
Not sure why this is happening; shouldn't [`Provider.tsx`](https://github.com/trentschnee/gatsby-theme-joy-ui/blob/576f43bb7ee4a2aa2dfd508d8f3c5b6f96ce1ca4/packages/gatsby-theme-joy-ui/src/theme/Provider.tsx#L12) be wrapping the page with `CssVarsProvider`?

**Reply by trentschnee** [reply]
Hey @alexkmnsky , sorry for this. This is why I should have had testing setup before publishing. I can confirm this is an issue. I am working on a fix now

**Reply by trentschnee** [reply]
@alexkmnsky I am **so sorry** about that! I have fixed everything and released version 1.2.0 now. You will need to pull the latest updates and run the command `yarn start` from within the root directory. Then you should be able to go to https://localhost:8000 and see the example.

Please let me know if you run into any other problems.

Thanks for your time and have a great day!

**Reply by kaminskyalexander** [reply]
Awesome, thanks for figuring that out. Running `yarn start` as you instructed works now. I will play around and see if I run into any other problems.

**Reply by kaminskyalexander** [reply]
Building isn't working for me, I've submitted a bug report #33 

Is static site generation unsupported?

**Reply by trentschnee** [reply]
> Building isn't working for me, I've submitted a bug report #33
> 
> Is static site generation unsupported?

Thanks for pointing this to my attention. I fixed the build issue and created serve / clean commands as well.
Note: Run `yarn clean` before building and serving
